### ISSUE TITLE
IndicesStatsService: Shard-level stats not included

### ISSUE THREAD

**Issue by mattayes** [olivere/elastic#1090]
Please use the following questions as a guideline to help me answer
your issue/question without further inquiry. Thank you.

### Which version of Elastic are you using?

[ ] elastic.v7 (for Elasticsearch 7.x)
[ ] elastic.v6 (for Elasticsearch 6.x)
[x] elastic.v5 (for Elasticsearch 5.x)
[ ] elastic.v3 (for Elasticsearch 2.x)
[ ] elastic.v2 (for Elasticsearch 1.x)

### Please describe the expected behavior

Using the [`IndicesStatsService`](https://godoc.org/github.com/olivere/elastic#IndicesStatsService) with [`level=shards`](https://godoc.org/github.com/olivere/elastic#IndicesStatsService.Level) would include individual shard-level statistics.

### Please describe the actual behavior

Individual shard-level statistics are missing from the [`IndexStats`](https://godoc.org/github.com/olivere/elastic#IndexStats) struct and shard-level info is missing.

### Any steps to reproduce the behavior?

```go
package main

import (
	"context"
	"fmt"
	"log"

	"gopkg.in/olivere/elastic.v5" // also an issue with other versions
)

func main() {
	client, err := elastic.NewClient()
	if err != nil {
		log.Fatal("Failed to create client: ", err)
	}
	defer client.Stop()

	index := "test"
	createIndex, err := client.CreateIndex(index).Do(context.Background())
	if err != nil {
		log.Fatal("Failed to create index: ", err)
	}
	if !createIndex.Acknowledged {
		log.Fatal("Index creation not acknowledged.")
	}
	defer func() {
		deleteIndex, err := client.DeleteIndex(index).Do(context.Background())
		if err != nil {
			log.Print("Failed to delete index: ", err)
		} else if !deleteIndex.Acknowledged {
			log.Print("Index deletion not acknowledged.")
		}
	}()

	resp, err := client.IndexStats("_all").Level("shards").Do(context.Background())
	if err != nil {
		log.Fatal("Failed to get index stats: ", err)
	}

	stats := resp.Indices[index]
	fmt.Printf("%v+\n", stats) // notice there are no shard stats
}
```

Running via curl, `IndexStats` looks something like this (assuming only one shard):

```
{
      "primaries" : {
        "store" : {
          "size_in_bytes" : 10112,
          "throttle_time_in_millis" : 0
        }
      },
      "total" : {
        "store" : {
          "size_in_bytes" : 20224,
          "throttle_time_in_millis" : 0
        }
      },
      "shards" : {
        "0" : [
          {
            "routing" : {
              "state" : "STARTED",
              "primary" : true,
              "node" : "U-lWmGsoQ6-aHhxJA3vXsw",
              "relocating_node" : null
            },
            "store" : {
              "size_in_bytes" : 162,
              "throttle_time_in_millis" : 0
            },
            "commit" : {
              "id" : "Q9MpBjJucg0L827RGxOL8A==",
              "generation" : 6,
              "user_data" : {
                "translog_uuid" : "iIQbj3aaRs-FZfJ4E8042g",
                "translog_generation" : "1",
                "max_unsafe_auto_id_timestamp" : "-1"
              },
              "num_docs" : 0
            },
            "shard_path" : {
              "state_path" : "/mnt/es/nodes/0",
              "data_path" : "/mnt/es/nodes/0",
              "is_custom_data_path" : false
            }
          },
          {
            "routing" : {
              "state" : "STARTED",
              "primary" : false,
              "node" : "9Yu_KxfETZaZSRmGojL2lA",
              "relocating_node" : null
            },
            "store" : {
              "size_in_bytes" : 162,
              "throttle_time_in_millis" : 0
            },
            "commit" : {
              "id" : "0kbuCQpRI+IPBvGxCWkl2g==",
              "generation" : 7,
              "user_data" : {
                "translog_uuid" : "DpbGOQUtTy63qWdj9ZPpig",
                "translog_generation" : "1",
                "max_unsafe_auto_id_timestamp" : "-1"
              },
              "num_docs" : 0
            },
            "shard_path" : {
              "state_path" : "/mnt/es/nodes/0",
              "data_path" : "/mnt/es/nodes/0",
              "is_custom_data_path" : false
            }
          }
        ]
      ]
    }
}

**Comment by olivere** [comment]
Thanks for the detailed bug report. I will tackle this in the next release. Unfortunately, I'm off for two weeks now so it'll take some time. Sorry for that.

**Comment by mattayes** [comment]
@olivere No worries, enjoy your vacation!

**Comment by olivere** [comment]
I think this is fixed by 1b8e368 and will soon be released in 5.0.81 (and ported to v6 and v7 as well).

**Comment by mattayes** [comment]
Thanks @olivere! Hope you had a good vacation.
